### ISSUE TITLE
Investigate our use of webfonts

### ISSUE THREAD

**Issue by diox** [mozilla/addons-frontend#10040]
A single request to AMO seems to cause 6 different requests to `.woff2` fonts, for a total of 635 KB transferred.
- Can we lower that ? Can we remove some or all of these fonts ?
- Should we even be using those fonts or follow https://protocol.mozilla.org/fundamentals/typography.html instead ?
- Should we [`font-display: swap`](https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face/font-display) to render more quickly without necessarily waiting for the fonts to download ?

**Comment by bobsilverberg** [comment]
I tried in both Firefox and Chrome and I only see one `.woff2` font file downloaded. Are some of the fonts perhaps pre-loaded in my browsers?

Also, how much, if any, UX input do we need on this issue?

**Comment by diox** [comment]
I was starting from a fresh cache.

![amo-fonts](https://user-images.githubusercontent.com/187006/106483946-61ee6580-64af-11eb-9479-bb95b3129d81.png)

(I've got the HAR export as well but it's not particularly interesting)

**Comment by diox** [comment]
The fonts are: Fira Sans Light, Fira Sans Regular, Fira Sans Medium, Fira Sans Bold, Open Sans Regular, Open Sans Bold. Open Sans is used in the footer, Fira for the rest of the page. The different weights used in the design cause us to use the different variants.

**Comment by eviljeff** [comment]
Mozilla websites use inter for the footer and main body text so we should switch to that for everything imo.  Just switching to inter would be a considerable transfer size saving as the file sizes are smaller.

(https://protocol.mozilla.org/fundamentals/typography.html details that we should maybe be using Zilla Slab* or Metropolis for headings but that's a bit more style adjustment work as we currently use Fira Sans for body and headings)

*The guide says "Mozilla" branded websites use Zilla Slab and "Firefox" branded websites use Metropolis, so I assume that'd be Metropolis for AMO with it's current logo (despite the domain name)

**Comment by willdurand** [comment]
> The fonts are: Fira Sans Light, Fira Sans Regular, Fira Sans Medium, Fira Sans Bold, Open Sans Regular, Open Sans Bold. Open Sans is used in the footer, Fira for the rest of the page. The different weights used in the design cause us to use the different variants.

We shouldn't change the fonts to something completely different without Product/UX sign-off.

The Photon font is Fira Sans. Open Sans is indeed used in the footer so I guess we can change that to use Fira Sans, it's not a major update.

I think we should start by using only one font (Fira Sans) and try to not have 3 or 4 different font formats and too many different variants (bold, regular, light, etc). This should help a bit without changing the UI too much until we hear from Product and UX. That is assuming the fonts are different (I haven't checked).

**Comment by eviljeff** [comment]
note: #10110 uses variable fonts - where all the variants can be calculated from a single file (which is bigger than one variant, but seemingly about double size - so an improvement when 3 or more are currently used).  Fira Sans doesn't appear to have a variable font variant (probably because it's 6 years old and that wasn't a thing at the time)

**Comment by willdurand** [comment]
> note: #10110 uses variable fonts - where all the variants can be calculated from a single file (which is bigger than one variant, but seemingly about double size - so an improvement when 3 or more are currently used). Fira Sans doesn't appear to have a variable font variant (probably because it's 6 years old and that wasn't a thing at the time)

Right, I believe we'll get a lot more value in adopting a new-ish font thanks to "variable fonts", etc. but changing a font is a major change. There are probably things we can do _before_ that, which is what I suggested above.

**Comment by willdurand** [comment]
> * Should we even be using those fonts or follow [protocol.mozilla.org/fundamentals/typography.html](https://protocol.mozilla.org/fundamentals/typography.html) instead ?

I am not sure. This is for marketing websites and it's unclear how that would work with Photon. I'd try to not change the fonts in the first place and see later if we still need to optimize fonts. (what @eviljeff did in a PR looks promising, although I don't know if that looks good visually/aesthetically speaking)

> * Should we [`font-display: swap`](https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face/font-display) to render more quickly without necessarily waiting for the fonts to download ?

Worth a try!

**Comment by eviljeff** [comment]
We can do these things:
- only use one font (drop Open Sans for the footer - use the same font as the page body).  Would save 2 files but only ~100kb because they're pretty small fonts.
- use variable fonts.  Fira Sans doesn't have variable fonts available so probably that means switching to a different font like Inter (unless there is some automagical way to generate a variable font file from the variants).  Saves the most - only one 220kb file is needed.
- define the css with `font-display: swap`.  Wouldn't save any download size but would (if the docs are correct) stop the fonts from blocking (at the cost, I guess, of potentiality rendering the page with the os default sans serif font for a moment during load).

**Comment by eviljeff** [comment]
![Inter-about](https://user-images.githubusercontent.com/768592/108836668-081e1e80-75c9-11eb-9074-406e008fad31.png)
![Inter-home](https://user-images.githubusercontent.com/768592/108836670-08b6b500-75c9-11eb-9bcb-2f8f6262cf69.png)
![Fira-about](https://user-images.githubusercontent.com/768592/108836663-06545b00-75c9-11eb-875b-9bc5239d12a7.png)
![Fira-home](https://user-images.githubusercontent.com/768592/108836666-06ecf180-75c9-11eb-83ff-3696e1f03f00.png)

**Comment by diox** [comment]
Maybe it's because I'm so used to how it looks now, but I think it looks kinda weird with Inter. Or not as good at least. I'm not a designer, UX or product person though, so it's not my call.

I do think we should ultimately switch away from Fira though: if it's not maintained because as you point out, it's holding us down by not supporting variable fonts.

**Comment by eviljeff** [comment]
@jvillalobos for input

**Comment by jvillalobos** [comment]
Can you please add a comparison of the two fonts for a listing page?

**Comment by eviljeff** [comment]
Inter
![Inter-detail](https://user-images.githubusercontent.com/768592/108886450-e476ca80-7600-11eb-9ca4-6f4e9551bc60.png)
Fira
![Fira-detail](https://user-images.githubusercontent.com/768592/108886445-e3de3400-7600-11eb-93a9-54e9132f39e9.png)

**Comment by Verdi** [comment]
We should be using Metropolis and Inter. Those screenshots above look fine except for the footer. Looks like Inter will need to be adjusted there.

**Comment by eviljeff** [comment]
Inter (with `font-weight: 400` applied, like protocol footers appear to have):

![Screen Shot 2021-03-11 at 13 48 09-fullpage](https://user-images.githubusercontent.com/768592/110639400-95f03100-81a7-11eb-81b0-26dfac7c968a.png)

**Comment by ioanarusiczki** [comment]
@eviljeff 

1. With a fresh profile - when I load the homepage there's only one `woff2` file loaded on AMO dev - compared to AMO stage where there are 6 requests with `font` as initiator. 
Let me know if that's alright. 

![only one woff](https://user-images.githubusercontent.com/33448286/111170009-2796de80-85ac-11eb-9a48-926b2ee99b5c.png)

2. All frontend pages have Inter Var as `font-family`. 

In 2 places, visually the Fira Sans font was more highlighted compared to what's with Inter. Let me know if that's fine. 

At the end of the footer:
on dev with Inter ->
![footer on dev](https://user-images.githubusercontent.com/33448286/111171388-8f99f480-85ad-11eb-827a-d3c87fb725bf.png)
on stage with Fira->
![footer on stage](https://user-images.githubusercontent.com/33448286/111171413-94f73f00-85ad-11eb-9342-faede8acdbdc.png)

The header:
on dev with Inter -> 
![header on dev](https://user-images.githubusercontent.com/33448286/111171634-c6700a80-85ad-11eb-8709-fa391aa6ecb3.png)
on stage with Fira->
![header on stage](https://user-images.githubusercontent.com/33448286/111171736-dbe53480-85ad-11eb-8e2b-e16bd4eef489.png)

**Comment by eviljeff** [comment]
@ioanarusiczki #10201 has been filed about the footer text (the header seems fine)
